.github/workflows/interop.yml

```diff
@@ -95,3 +95,28 @@ jobs:
           go run testautonatv2.go &
           cd ../nim-peer
           nim r src/nim_peer.nim $(cat ../go-peer/peer.id)
+
+  run-kad-interop:
+    name: Run Kademlia DHT interoperability tests
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Set up Nim
+        uses: jiro4989/setup-nim-action@v1
+        with:
+          nim-version: "stable"
+
+      - name: Run Nim and Rust together
+        run: |
+          nimble install
+          cd interop/kad/rust-peer
+          cargo run &
+
+          cd ../nim-peer
+          nim r src/nim_peer.nim $(cat ../rust-peer/peer.id)
```

Comment:
why ` $(cat ../rust-peer/peer.id)` ?

Reply:
First argument is the peerID of the peer you're talking to. Rust-peer saves its peer id in that file.

Reply:
later in nim code there is this:
```
peerId = PeerId.init(readFile("../rust-peer/peer.id")).get()
```